kuma/javascript/src/document.jsx

```diff
@@ -314,41 +223,132 @@ function Content({ document }: DocumentProps) {
     );
 }
 
-export default function Page() {
-    const document = useContext(DocumentProvider.context);
-    const ga = useContext(GAProvider.context);
+function DocumentPage({ document }: DocumentProps) {
+    return (
+        <>
+            <Header document={document} />
+            <TaskCompletionSurvey document={document} />
+            <Titlebar title={document.title} document={document} />
+            <Breadcrumbs document={document} />
+            <Content document={document} />
+        </>
+    );
+}
 
-    /*
-     * Register an effect that runs every time we see a new document URL.
-     * The effect sends a Google Analytics timing event to record how long
-     * it took from the start of the navigation until the new document is
-     * rendered. And then it turns off the loading indicator for the page
-     *
-     * Effects don't run during server-side-rendering, but that is fine
-     * because we only want to send the GA event for client-side navigation.
-     * Calling navigateRenderComplete() will have no effect if
-     * navigateStart() was not previously called.
-     */
-    useEffect(() => {
-        navigateRenderComplete(ga);
+type Props = { data: ?DocumentData };
```

Comment:
Haven't seen this syntax before - `?` before the type. What's up with that? (Searching the internet was fruitless. 😄)

Reply:
The ? means optionally null in Flow.

Reply:
Ah! I assumed it was something to do with nullable/optional values.

(I also had it in my head that you were using TypeScript, hence my internet search failure.)